README.md

```diff
@@ -1,3 +1,270 @@
+Just populate the config struct with the default values and pass the struct to Dials. 
+Dials also allows the flexibility to choose the precedence order to determine which sources can overwrite the configuration values. Additionally, Dials has special handling of structs that implement [`encoding.TextUnmarshaler`](https://golang.org/pkg/encoding/#TextUnmarshaler) so structs (like [`IP`](https://pkg.go.dev/net?tab=doc#IP) and [`time`](https://pkg.go.dev/time?tab=doc#Time)) can be properly parsed.
+
+## Using Dials
+
+### Reading from config files, environment variables, and command line flags
+Dials requires very minimal configuration to populate values from several sources through the `ez` package. Define the config struct and provide a method `ConfigPath() (string, bool)` to indicate the path to the config file. The package defines several functions that help populate the config struct by reading from config files,
+environment variables, and command line flags
+
+``` go
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/vimeo/dials/ez"
+)
+
+// Config is the configuration struct needed for the application
+type Config struct {
+	// When a struct tag more specifically corresponding to a source is
+	// present, it takes precedence over the `dials` tag. Note that just
+	// because there is a `yaml` struct tag present doesn't mean that other
+	// sources can't fill this field.
+	Val1 string `dials:"Val1" yaml:"b"`
+	// the dials tag can be used as an alias so when the name in the config file
+	// changes, the code doesn't have to change.
+	Val2 int `dials:"val_2"`
+	// Dials follows the Go convention for flags and will look for the dials
+	// tag or field name in lower-kebab-case. Without any struct tags, it
+	// would look for val-3 flag. To specify a different flag name, use the
+	// `dialsflag` tag. Now, Dials will lookup "some-val" flag instead.
+	// Since only `dialsflag` tag is specified, Val3 will only be populated
+	// from command line flags
+	Val3 bool `dialsflag:"some-val"`
+	// Path holds the value of the path to the config file. Dials follows the
+	// Go convention and will look for the dials tag or field name in all caps
+	// when struct tags aren't specified. Without any struct tags, it would
+	// lookup the PATH environment variable. To specify a different env variable,
+	// use the `dials_env` tag. Now Dials will lookup "configpath" env value to
+	// populate the Path field
+	Path string `dials_env:"configpath"`
+}
+
+// ConfigPath returns the path to the config file. This is particularly helpful
+// when the path is populated from environment variables or command line flags.
+// Dials will first read from environment variables and command line flags
+// and then read the config file specified by the populated field.
+func (c *Config) ConfigPath() (string, bool) {
+	// can alternatively return empty string and false if the state of the
+	// struct doesn't specify a config file to read
```

Comment:
Hmm, users who just want to use environment variables and flags probably don't need the `ez` package's helpers. (it might actually be more confusing to use `ez` than not.)

Perhaps we should have a simple example of just the flag source and environment variable source being passed to dials. (definitely in another section, though.)